### DISCUSSION TITLE
How many of you are using admob-plus right now?

### DISCUSSION THREAD

**Discussion by Nuesret** [admob-plus/admob-plus#694] [Q&A]
After the problems I have with implementation and the fact that there is little or no help to be found on the web, I assume that very few people are using the plugin. Most of them will most likely have had the same experience as me. That they try and ultimately fail. So I have no choice but to implement Admob directly in Android Studio and do without the plugin. How many of you have successfully used the plugin? I mean with full control in terms of showing and hiding? I guess not many.

**Comment by 1n3JgKl9pQ6cUMrW** [comment]
This is one of the best and most used Cordova Admob plugins there is, with sometimes even more than 500 active downloads and installs per week.

![image](https://github.com/user-attachments/assets/9700bca6-e4f7-4df1-81f7-742b71b51076)

For me personally, I use it (and it's predecessors) for about 10 years without any big issues.

Development is done slowly, but steady and I am fine with that.

And if you don't like it, create it yourself or use the Community Fork.

And "yes", showing and hiding is no problem. You can't destroy the instance, but you can hide it and re-generate afterwards.

**Comment by Nuesret** [comment]
Hello, thank you very much for your answer. Please don't get me wrong: I am very grateful for this plugin. I really am. I just can't seem to get it to work completely. It's exasperating. I mean, I manage to get at least one banner to show everywhere - but capitulate when I want it to disappear and reappear. If you've been using this for 10 years - you obviously know your way around the plugin really well. Could you help me find the mistakes I'm making? I would also pay you something. email@kaymak.de

**Reply by hooliapps** [reply]
Hello I use this plugin on cordova with 0 problems.

**Reply by 1n3JgKl9pQ6cUMrW** [reply]
> Hello I use this plugin on cordova with 0 problems.

There are some problems, like for the Native Ad that scrolls "on top" of the content. But that's due to the implementation.

I solved it to show the Native Ad fixed, so when you scroll the content, the ad stays fixed like it's inside a modal.

And I injected a "close ad" button below the ad, so you can hide / close it (like a real  modal).

Also the Anchored Inline Banners (the newer version of the deprecated fixed banners) are not truly responsive when rotating from landscape to portrait (I fixed this with a fork, that recalculates the DPI and PPI).

Both are not 100%, but for things like the regular banner, reward and interstitial it's very, very good.

**Comment by 1n3JgKl9pQ6cUMrW** [comment]
I can't help you with your specific code / problem.

But for me `banner.hide();` and `banner.show();` do work as intended.

![image](https://github.com/user-attachments/assets/4adb00c1-c180-4b41-841b-f5fe14c2a63d)

**For the record;** I do use the "old" version of this plug-in, not the bèta version (since it's bèta and not overall stable).

Also I use the `Community version` of this plug-in, which is the same as the "old" one, but more up to date with the `SDK's`.

https://github.com/EYALIN/community-admob-plus

I only work with Cordova / Android and all my apps are SPA : single page application (`index.html`).

I only have one banner, which I can show, or hide. When I try to ad another one, it's stacked on top of the previous banner (which I don't want).

If things are not working for you, consider another plugin, based (in the mindset) on the one you are using right now;

https://github.com/EMI-INDO/emi-indo-cordova-plugin-admob

---

_P.S. Yeah, it's quit silent here in the discussion topic._

**Comment by Nuesret** [comment]
SPA... I see. And a single banner. Ok. That's what I'm doing now (had 15 unitIDs before) 

I'm really pleased that someone has come forward. Thank you once again.

On last thing... 

do you initialize the plugin (the plugin DeviceReady code) in MainAcitivity or are you using this admob code from the Google Developers page 
```
new Thread(
            () -> {
              // Initialize the Google Mobile Ads SDK on a background thread.
              MobileAds.initialize(this, initializationStatus -> {});
            })
        .start();
```
in MainActivity.java ?

I suppose, you're using modals and iframes in your SPA? I'm also using pages with modals. And my intention was hiding the banner on one certain modal with Youtube-video. But the hidebanner function did not work. So I created a iframe and later a seperate html page which I wanted to use as a overlay with a background-color but they didn't work either.

Is there a possibility in javascript to force a banner to be invisible? Maybe with a hack to position the banner outside the screen an d hide it with overflow: hidden?

**Comment by 1n3JgKl9pQ6cUMrW** [comment]
I have those ads;

* Native
* Interstitial
* Banner
* Reward

Each (of course) with it's own #ID.

But I have only one instance of each, not multiple.

But you can request as many reward-ads as you want, for example.

One sees the reward, closes it, get rewarded and can request a new reward-ad, using the same ID.

And banner you can show, or hide withing the same ID (it's updating every minute or so to show another ad).

You can bind a single ID to the internal ID of the ad (which is set by Google) and see that in the stats of things like Firebase / Google Analytics.

So for me, it's working : I don't want multiple unique banners, one is enough for me (and I can show it, hide it and "read" the stats based on it's internal ID).

---

Regarding you Java-code, I don't write, use or even touch Java-code.

I am writing all my apps as HTML-apps, and debug them in my browser. And I use Cordova to wrap in into a real app (the APK-file), so I can communicate with some basic Android API's.

I don't use Android Studio, or something like a "real" IDE. I work in Visual Studio Code, with is like a text-editor on steroids.

I simply start a new Cordova project, add the Admob plugin (the latest stable version) and initialize it on `deviceready`. That's it for me.

All debugging and development is done in the browesr; just hit F12 to see the console, set a responsive size for your "app" and away you go.

**Be aware : AdMob will NOT work in the browser; you need the "real" Android app for that part (on an emulator or real device).**

```
let banner

document.addEventListener('deviceready', async () => {
  await admob.start()

  banner = new admob.BannerAd({
    adUnitId: 'ca-app-pub-xxx/yyy',
  })

  await banner.show()
}, false)
```

_(of course my code is more complex, but in the basic this is it)_

That's it, I never do anything using Java (the above is Javascript, not Java).

O, I don't use iFrames (but can embed Youtube with the default HTML-code, which will show the movie inside an Iframe). And I use Bootstrap for the styling-framework, that comes with modals.

```
<iframe width="1280" height="720" src="https://www.youtube.com/embed/sQbwqc5biSc" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe>
```

When I add this in my page, it's shown in the app. Afterall Cordova is just HTML / CSS and JS.

**Comment by Nuesret** [comment]
Thank you for the explanations. Yes, I understand. It's the same for me: I also develop with html-css-javascript in VSC and compile the result in cordova, where I insert the admob plugin. Then I import the project into Android Studio and edit it there - exactly as you described. However, I only use banners. I have also compiled native apps in Java in AS - but I haven't worked much with the admob implementation yet. I'm currently in the process of learning it - because the structure and implementation are quite simple. Like VSC, AS also has a preview function next to it in the layout window, where you can see the changes simultaneously. Testing on a real device is also lightning fast. My tendency is already towards native apps in AS - although I would definitely also like to integrate Javascript. There are also a number of Java animation options - but I don't want to do without the many great js libraries.

**Reply by 1n3JgKl9pQ6cUMrW** [reply]
What's AS, Action Script?

And why import the project into Android Studio?

You can compile all from the terminal / command line, no need for Studio :-)

Best is to start a simpel app, just `hello world`, add one banner in it and a button that does `show` and `hide` it.

When that's working, extend to the full app step by step.

Most important tip : when writing Cordova apps, 99,99% of us don't need Java or Android Studio.

After all, it's plain non-compiled text you are coding, with some plug-ins that hook into the real Java (API) environment.

You can even open your compiled .APK into a thing like 7ZIP, extract it, et voila... the `www` folder and all the assets are there.

When done good, you app will even run from that unzipped folder, just by double clicking on it.

I mis-abuse this feature even as a kind of version control, since I'm not very strict with .git :-)

**Comment by Nuesret** [comment]
AS = Android Studio. But I used a small amount of Action Script in Flash projects years ago. - till Steve Jobs killed the flash plugin. Of course you don't have to use Android Studio to compile apps. But I am also very interested in using it. At the moment I still use VSC and cordova CLI as a basis and make additions in AS. But I consider engines like Ct.js, Godot, renpy or Defold - from which you can compile apps directly - to be very interesting.
